src/main/sensors/boardalignment.c

```diff
@@ -59,28 +61,59 @@ void initBoardAlignment(const boardAlignment_t *boardAlignment)
     rotationAngles.angles.pitch = degreesToRadians(boardAlignment->pitchDegrees);
     rotationAngles.angles.yaw   = degreesToRadians(boardAlignment->yawDegrees  );
 
-    buildRotationMatrix(&rotationAngles, boardRotation);
+    buildRotationMatrix(&rotationAngles, &boardRotation);
 }
 
-static void alignBoard(float *vec)
+FAST_CODE static void applyRotation(float *vec, fp_rotationMatrix_t *rotationMatrix)
 {
     float x = vec[X];
     float y = vec[Y];
     float z = vec[Z];
 
-    vec[X] = (boardRotation[0][X] * x + boardRotation[1][X] * y + boardRotation[2][X] * z);
-    vec[Y] = (boardRotation[0][Y] * x + boardRotation[1][Y] * y + boardRotation[2][Y] * z);
-    vec[Z] = (boardRotation[0][Z] * x + boardRotation[1][Z] * y + boardRotation[2][Z] * z);
+    vec[X] = (rotationMatrix->m[0][X] * x + rotationMatrix->m[1][X] * y + rotationMatrix->m[2][X] * z);
+    vec[Y] = (rotationMatrix->m[0][Y] * x + rotationMatrix->m[1][Y] * y + rotationMatrix->m[2][Y] * z);
+    vec[Z] = (rotationMatrix->m[0][Z] * x + rotationMatrix->m[1][Z] * y + rotationMatrix->m[2][Z] * z);
+
+}
+
+FAST_CODE static void alignBoard(float *vec)
+{
+    applyRotation(vec, &boardRotation);
+}
+
+FAST_CODE void alignSensorViaMatrix(float *dest, fp_rotationMatrix_t* sensorRotationMatrix)
+{
+    applyRotation(dest, sensorRotationMatrix);
+
+    if (!standardBoardAlignment)
+        alignBoard(dest);
+}
+
+FAST_CODE void alignSensorViaMatrixFromRotation(float *dest, uint8_t rotation)
```

Comment:
This code is for unitest only ? It should probably be #ifdef'ed

Reply:
as noted, cleanup on this PR has not been done yet, it's an in-progress refactor to get clarification from @mikeller if the direction of this PR is correct - @mikeller confirmed via slack it is what he wants, so watch for cleanup commits on this PR to address this.